tests/samplers_tests/test_qmc.py

```diff
@@ -0,0 +1,425 @@
+from collections import OrderedDict
+import sys
+from typing import Any
+from typing import Callable
+from typing import Dict
+from unittest.mock import Mock
+from unittest.mock import patch
+import warnings
+
+import numpy as np
+import pytest
+
+import optuna
+from optuna.distributions import BaseDistribution
+from optuna.trial import Trial
+from optuna.trial import TrialState
+
+
+_SEARCH_SPACE = OrderedDict(
+    {
+        "x1": optuna.distributions.IntUniformDistribution(0, 10),
+        "x2": optuna.distributions.IntLogUniformDistribution(1, 10),
+        "x3": optuna.distributions.UniformDistribution(0, 10),
+        "x4": optuna.distributions.LogUniformDistribution(1, 10),
+        "x5": optuna.distributions.DiscreteUniformDistribution(1, 10, q=3),
```

Comment:
```suggestion
        "x1": optuna.distributions.IntDistribution(0, 10),
        "x2": optuna.distributions.IntDistribution(1, 10, log=True),
        "x3": optuna.distributions.FloatDistribution(0, 10),
        "x4": optuna.distributions.FloatDistribution(1, 10, log=True),
        "x5": optuna.distributions.FloatDistribution(1, 10, step=3),
```

This change should be applied together with https://github.com/optuna/optuna/pull/2423/files#diff-f6e75a1153a00e6980c0601537c46273706ea852e87df915ccc913268d1e8063R232.